packages/ts/react-crud/test/FormController.ts

```diff
@@ -8,20 +8,21 @@ export type FormElement = HTMLElement & {
   checked?: boolean;
 };
 
+type FormQueries = Pick<RenderResult, 'findByLabelText' | 'findByTestId' | 'findByText'>;
+
 export default class FormController {
   readonly instance: HTMLElement;
-  readonly #result: RenderResult;
+  readonly #result: FormQueries;
   readonly #user: ReturnType<(typeof userEvent)['setup']>;
 
-  static async init(result: RenderResult, user: ReturnType<(typeof userEvent)['setup']>): Promise<FormController> {
-    const form = (await waitFor(
-      () => result.container.querySelector('vertical-layout[theme="padding"]')!,
-    )) as HTMLElement;
+  static async init(result: FormQueries, user: ReturnType<(typeof userEvent)['setup']>): Promise<FormController> {
+    const form = await waitFor(async () => result.findByTestId('auto-form'));
```

Comment:
@taefi used search by `vaadin-form-layout`. What do you think about using it instead of `data-testid`?

Reply:
There is no `vaadin-form-layout` at this point. From a testing best practices POV it doesn't make a difference as the user interacting with the app doesn't care about what kind of element is rendered. I'd suggest to leave as is, I'm fine with it being changed in a subsequent PR.